File: app/src/main/java/com/friendlypos/login/modelo/UserError.java
package com.friendlypos.login.modelo;

import com.google.gson.Gson;

import java.io.IOException;

import okhttp3.ResponseBody;

/**
 * Created by DelvoM on 25/09/2017.
 */

public class UserError {

    private String error;
    private String message;

    public UserError(String error, String message) {
        this.error = error;
        this.message = message;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public static UserError fromResponseBody(ResponseBody responseBody) {
        Gson gson = new Gson();
        try {
            return gson.fromJson(responseBody.string(), UserError.class);
        } catch (IOException e) {
            e.printStackTrace();
        }

        return null;
    }
}

File: app/src/main/java/com/friendlypos/login/modelo/UserError.kt
package com.friendlypos.login.modelo

import com.google.gson.Gson
import okhttp3.ResponseBody
import java.io.IOException

/**
 * Created by DelvoM on 25/09/2017.
 */
class UserError(var error: String, var message: String) {
    companion object {
        fun fromResponseBody(responseBody: ResponseBody): UserError? {
            val gson: Gson = Gson()
            try {
                return gson.fromJson<UserError>(responseBody.string(), UserError::class.java)
            } catch (e: IOException) {
                e.printStackTrace()
            }

            return null
        }
    }
}
